packages/expo-modules-core/src/web/CoreModule.ts

```diff
@@ -33,23 +40,36 @@ class EventEmitter<TEventsMap extends EventsMap> implements EventEmitterType {
     eventName: EventName,
     listener: TEventsMap[EventName]
   ): void {
-    this.listeners?.get(eventName)?.delete(listener);
+    const hasRemovedListener = this.listeners?.get(eventName)?.delete(listener);
+    if (this.listenerCount(eventName) === 0 && hasRemovedListener) {
+      this.stopObserving(eventName);
+    }
   }
 
   removeAllListeners<EventName extends keyof TEventsMap>(eventName: EventName): void {
+    const listenerCount = this.listenerCount(eventName);
+
     this.listeners?.get(eventName)?.clear();
+    if (listenerCount > 0) {
+      this.stopObserving(eventName);
+    }
```

Comment:
```suggestion
    const previousListenerCount = this.listenerCount(eventName);
    this.listeners?.get(eventName)?.clear();
    if (previousListenerCount > 0) {
      this.stopObserving(eventName);
    }
```
Let's rename it to make it more readable.